### PR TITLE
goal errors are cleared at the beginning of the next command

### PR DESCRIPTION
**Pull request by AstroStucky**
This PR spawned from discussion in #361

## Summary of Changes
* Reset the relevant action group GOAL_ERROR flag at the beginning of an action. 

## Test
Launch any simulator world
1. Monitor the system faults topic `rostopic echo /system_faults_status` and the arm faults topic `rostopic echo /arm_faults_status` in two separate terminals. 
2. Call an invalid ArmMoveCartesian `rosrun ow_lander arm_move_cartesian.py`
3. The command will fail. Both topics should show a value of 2. 
4. Call a valid TaskGrind `rosurn ow_lander task_grind.py`
5. The system fault will remain 2, because we are calling an action in a different group (TASK), but arm faults will be reset to 0 at the beginning of the TaskGrind because it uses the arm. Let the TaskGrind complete.
6. Call a valid ArmMoveCertaisn `rosrun ow_lander arm_move_cartesian.py -x 1.8 -z 0.5`
7. System faults will become 0 immediately. The action will complete and both topics should both now be 0.

### CODE REVIEW

**Comment by kmdalal** [comment]
Unrelated to this PR, but the default invocations of `arm_move_cartesian.py` and all the action scripts should be valid, and we should make this so for Release 12...

File: ow_lander/src/ow_lander/server.py [hunk 1]
```diff
@@ -138,6 +138,14 @@ def __on_action_called(self, goal):
                    "This should never happen!")
       return
     rospy.loginfo(f"{self.name} action started")
+    # reset execution fault flag by sending a PENDING state
```
Comment (kmdalal) [inline ow_lander/src/ow_lander/server.py:141]: Is the rest of this paragraph consistent with this first sentence, which says PENDING state?

**Review by kmdalal — APPROVED** [review]
The given test passes, but I didn't do any more.  I don't fully understand the logic of the code change, but it sounds reasonable so will leave it to your judgement.

**Comment by AstroStucky** [comment]
> Unrelated to this PR, but the default invocations of `arm_move_cartesian.py` and all the action scripts should be valid, and we should make this so for Release 12...

The reason ArmMoveCartesian, ArmMoveCartesianGuarded, and other actions do not have zero XYZ defaults is because:
    **A)** From the user perspective, there is not a whole lot of use in a default move cartesian position. If you use the action, there is a presumption that you have a very particular pose in mind you would like the arm to go to. We, as developers and testers of the actions, are perhaps the only ones that use it without much care for where the arm winds up. 
    **B)** It makes the use of the relative flag harder. Currently if you merely want to move downward by 20cm it is as simple as `rosrun ow_lander arm_move_cartesian.py -r -z -0.2`. If there are defaults for x, y, and z, you would have to explicitly state which coordinates should have zero translation `rosrun ow_lander arm_move_cartesian.py -r -x 0 -y 0 -z -0.2`

It's my opinion that the actions are more useful to the user when X, Y, Z default to zero. Keep in mind, this is only particular to how the action client script is defined and has nothing to do with the action definition itself. We could discuss this further and come up with a different argparse interface that still satisfies the usability of **B**, but I still think my point in **A** is valid and correct.

File: ow_lander/src/ow_lander/server.py (hunk 1 above)
Comment (AstroStucky) [reply ow_lander/src/ow_lander/server.py:141]: Whoops. I originally planned to reset it with a PENDING state. It does not ultimately matter given how ActionGoalStates works. I

Good catch. This is fixed.

Comment (AstroStucky) [inline ow_lander/src/ow_lander/server.py:141]:
```suggestion
    # reset execution fault flag by sending a SUCCEEDED state
```

**Pull request merged**